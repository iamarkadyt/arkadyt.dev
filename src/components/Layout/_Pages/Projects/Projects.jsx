import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';

import Card from '../../Card/Card';

import classes from './Projects.module.css';
import projects from '../../../../projects.content';
import { sentenceToURL } from '../../../../utils/utils';

export class Projects extends React.Component {
    routes = [];

    render() {
        return <div className={classes.container}>
            <Switch>
                {/* When testing, separate set of data is passed through props,
            to maintain snapshot integrity. */}
                {(this.props.data || projects).map((item, index) => {
                    const route =
                        `${this.props.match.url}/${sentenceToURL(item.header)}`;
                    this.routes.push(route);
                    return <Route key={index} path={route}
                        render={props => <Card {...props} {...item} />} />
                })}
                <Redirect to={this.routes[0]} />
            </Switch>
        </div>;
    }
}

export default Projects;