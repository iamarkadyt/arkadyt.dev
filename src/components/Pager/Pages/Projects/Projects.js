import React, { Component } from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';

import ProjectCard from './ProjectCard/ProjectCard';
import NavBar from './NavBar/NavBar';

import classes from './Projects.module.css';
import data from '../../../../projects.content';

/**
 * /projects route.
 * Presents projects as cards, one per route at local router.
 * Also has own NavBar for its router.
 */
class Projects extends Component {
    getData = () => {
        // When testing, separate set of data is passed through props,
        // to maintain snapshot integrity.
        return this.props.data || data;
    }

    render() {
        const routes = [];

        return <div className={classes.container}>
            <Switch>
                {this.getData().map((blob, index) => {
                    const route =
                        `${this.props.match.url}/${blob.header.replace(' ', '-').toLowerCase()}`;
                    routes.push(route);

                    return <Route key={index} path={route}
                        render={() => <ProjectCard header={blob.header}
                            img={blob.img}
                            url={blob.url}>
                            {blob.body}
                        </ProjectCard>} />
                })}
                <Redirect to={routes[0]} />
            </Switch>
            <NavBar urls={routes} className={classes.navBar} />
        </div>;
    }
}

export default Projects;