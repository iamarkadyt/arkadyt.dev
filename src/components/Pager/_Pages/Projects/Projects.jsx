import React from 'react';
import { connect } from 'react-redux';
import * as actionTypes from '../../../../store/actions';
import { Route, Redirect, Switch } from 'react-router-dom';

import Card from '../../Card/Card';

import classes from './Projects.module.css';
import data from '../../../../projects.content';

/**
 * /projects route.
 * Sets up routes individually for every project.
 */
class Projects extends React.Component {
    routes = [];

    render() {
        return <div className={classes.container}>
            <Switch>
                {/* When testing, separate set of data is passed through props,
            to maintain snapshot integrity. */}
                {(this.props.data || data).map((blob, index) => {
                    const route =
                        `${this.props.match.url}/${blob.header.replace(' ', '-').toLowerCase()}`;
                    this.routes.push(route);

                    return <Route key={index} path={route}
                        render={() => <Card header={blob.header}
                            img={blob.img}
                            url={blob.url}
                            body={blob.body} />
                        } />
                })}
                <Redirect to={this.routes[0]} />
            </Switch>
        </div>;
    }

    componentDidMount() {
        this.props.onProjectsFetched(this.routes);
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onProjectsFetched: routes => dispatch({ type: actionTypes.UPDATE_ROUTES, payload: routes })
    }
}

export default connect(null, mapDispatchToProps)(Projects);