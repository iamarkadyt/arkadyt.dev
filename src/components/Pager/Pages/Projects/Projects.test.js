import React from 'react';
import Enzyme, { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import Projects from './Projects';

configure({ adapter: new Adapter() });

const data = [
    {
        header: 'Area 55',
        url: 'http://a.com',
        img: 'https://picsum.photos/200/300/?random',
        body: 'Et dolore qui eiusmod ullamco ipsum nisi culpa in irure fugiat. Sint eiusmod reprehenderit anim cupidatat cupidatat aliquip esse. Cillum magna occaecat sunt cupidatat nisi magna nisi. Incididunt magna esse cupidatat reprehenderit irure et consectetur cillum. Nisi exercitation consequat quis non aliqua aliqua id excepteur velit qui consectetur cillum adipisicing.'
    }
];

it('matches snapshot.', () => {
    const tree = renderer.create(
        <Router>
            {/* '/' causes match and following render */}
            <Route path='/' 
                render={(props) => <Projects data={data} {...props} />} />
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});