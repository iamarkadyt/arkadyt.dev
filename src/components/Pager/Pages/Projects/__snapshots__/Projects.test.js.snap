// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot. 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <p>
      <span
        className="header"
      >
        Area 55
      </span>
      <br />
      <br />
      Et dolore qui eiusmod ullamco ipsum nisi culpa in irure fugiat. Sint eiusmod reprehenderit anim cupidatat cupidatat aliquip esse. Cillum magna occaecat sunt cupidatat nisi magna nisi. Incididunt magna esse cupidatat reprehenderit irure et consectetur cillum. Nisi exercitation consequat quis non aliqua aliqua id excepteur velit qui consectetur cillum adipisicing.
      <br />
      <br />
      <a>
        http://a.com
      </a>
    </p>
    <img
      alt=""
      src="https://picsum.photos/200/300/?random"
    />
  </div>
  <ul
    className="list"
  >
    <li>
      <a
        aria-current="true"
        className="active"
        href="//area-55"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m37.3 20q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
            />
          </g>
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current={false}
        className={undefined}
        href="//area-71"
        onClick={[Function]}
        style={undefined}
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m37.3 20q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
            />
          </g>
        </svg>
      </a>
    </li>
  </ul>
</div>
`;
